### PR TITLE
Add test to BookService, CategoryService, BookController and Category…

### PR DESCRIPTION
**Pull request by AntonBabychP1T**
Screenshot 2023-12-22 115314
…Controller

### CODE REVIEW

**Review by Rommelua — CHANGES_REQUESTED** [review]
Java Cl checks failed. Before making a Pull Request (or a new commit to an already existing PR), you need to write the mvn clean package command in the terminal and make sure that all checks pass. After that, the mentor will be able to review the homework

image

File: src/test/java/store/bookstoreapp/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+import store.bookstoreapp.dto.book.BookDto;
+import store.bookstoreapp.dto.book.CreateBookRequestDto;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerTest {
+
```
Comment (Sanderfuck) [inline src/test/java/store/bookstoreapp/controller/BookControllerTest.java]:
```suggestion
```

File: src/test/java/store/bookstoreapp/controller/BookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        thirdBook.setAuthor("Third Author");
+        thirdBook.setTitle("Third title");
+        thirdBook.setPrice(BigDecimal.valueOf(19.99));
+        thirdBook.setIsbn("3ISBN03");
+        thirdBook.setDescription("description");
+        thirdBook.setCoverImage("coverimage");
+        List<BookDto> expected = new ArrayList<>();
+        expected.add(firstBook);
+        expected.add(secondBook);
+        expected.add(thirdBook);
```
Comment (Sanderfuck) [inline src/test/java/store/bookstoreapp/controller/BookControllerTest.java]: Create templates of entities and reuse it in every test what you need

File: src/test/java/store/bookstoreapp/service/BookServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Mock
+    private BookRepository bookRepository;
+
+    @Mock
+    private BookMapper bookMapper;
+    @Mock
+    private BookSpecificationBuilder bookSpecificationBuilder;
```
Comment (Sanderfuck) [inline src/test/java/store/bookstoreapp/service/BookServiceTest.java]: Manage your empty lines! Inject place after Mock

File: src/test/java/store/bookstoreapp/service/BookServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+        List<Book> books = List.of(book);
+        Page<Book> bookPage = new PageImpl<>(books, pageable, books.size());
+
+        when(bookRepository.findAll(pageable)).thenReturn(bookPage);
+        when(bookMapper.toDto(book)).thenReturn(bookDto);
+
+        //when
+        List<BookDto> bookDtos = bookService.findAll(pageable);
+
+        //then
```
Comment (Sanderfuck) [inline src/test/java/store/bookstoreapp/service/BookServiceTest.java]: Don`t needs comment in this case

File: src/test/java/store/bookstoreapp/service/CategoryServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+import store.bookstoreapp.exception.EntityNotFoundException;
+import store.bookstoreapp.mapper.CategoryMapper;
+import store.bookstoreapp.model.Category;
+import store.bookstoreapp.repository.category.CategoryRepository;
+import store.bookstoreapp.service.impl.CategoryServiceImpl;
+
+@ExtendWith(MockitoExtension.class)
+public class CategoryServiceTest {
+
+    private static final String VALID_NAME = "Valid name";
```
Comment (Sanderfuck) [inline src/test/java/store/bookstoreapp/service/CategoryServiceTest.java:33]: Keep one style of empty line after naming class

**Review by Sanderfuck — CHANGES_REQUESTED** [review]
image
Fix your tests at first

**Comment by Sanderfuck** [comment]
Fix your checks!!!!

**Pull request merged**